### PR TITLE
Related to Issue #342 - Tag key differentation when a supporting value can or cannot exist.

### PR DESCRIPTION
**Pull request by cnharris10**
Related to Issue: https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/342

Further clarifying that:
1. Tag keys that cannot support values are given a `true` (boolean) value
--> (ex: [GCP single-entity tags](https://cloud.google.com/vpc/docs/add-remove-network-tags)) `vpc` becomes `"vpc":true` and is included in `Tags`
3. Tag keys that exist but aren't set on a resource aren't included as tags
--> If tag `team` exists but is not set on resource id 'abc', associated `Tags` will _not_ contain `"team":true`

### CODE REVIEW

**Comment by rileyjenk** [comment]
Are we saying that {"name":"peter", "version":NULL} and  {"name":"peter" } are equivalent. My concern here is that we are disagreeing with most technologies/languages that say otherwise. I would say we allow either as both are valid and let the provider decide on which they follow.

Single value tags are not key value tags and often function as default true value tags. Can we specify that?
Example: 
['unicorns','dragons','UFOs'] translates to {'unicorns':TRUE, 'dragons':TRUE, 'UFOs':True'}.

**Comment by cnharris10** [comment]
@rileyjenk so your suggestions are:

1. Change value-only tags to have "true" as a default value (instead of null)
2. Do not explicitly restrict providers (maybe say SHOULD NOT) from providing k/v tags that would have a null value (even though they won't anyways)

Is this correct?

**Comment by rileyjenk** [comment]
@cnharris10 yes that is correct.

File: specification/columns/tags.md
Comment (cnharris10) [reply specification/columns/tags.md:12]: Correct. For example, if an account has 100 total tags created across 10 resources, the only tags that should be included with a resource's line item are the N tags associated with that resource - not all 100 tags.

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [inline specification/columns/tags.md:13]: I'm good with line 13.

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [inline specification/columns/tags.md:12]: I'm mostly good with line 12. It gives freedom to the cloud provider to decide what tags to include. Here is a suggested rewording to make it more clear the tag should have a non-null value and what should be included:

> A Tag key-value pair where the value is non-null for a given resource SHOULD be included in the tags column.

This also may raise the question to the reader "What about tag keys that have null values?". We could add a line that maintains the freedom for cloud providers:

> A Tag key-value pair where the value is null for a given resource MAY be included in the tags column, depending on the provider's tag finalization process.

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [inline specification/columns/tags.md:29]: Shall we expand the example to include examples of:
1. labels
2. tags with null values being optionally included

File: specification/columns/tags.md
Comment (cnharris10) [reply specification/columns/tags.md:12]: I like both but just without `-value pair` because it is defined as a KeyValueFormat attribute already.

Does this edits work? @AWS-ZachErdman 

```
A Tag key where its value is non-null for a given resource SHOULD be included in the tags column.
A Tag key where its value is null for a given resource MAY be included in the tags column, depending on the provider's tag finalization process.
```

Comment (AWS-ZachErdman) [reply specification/columns/tags.md]: looks good to me!

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [inline specification/columns/tags.md:25]: > The third tag with tag key, `baz`, does not support a value, so `null` is assigned as its value.

If it does not *support* a value, does that mean it is label and should be "true"?

My semantics may be wrong, but to me, not *supporting* a value means "label", whereas not *having* a value means "null".

**Review by AWS-ZachErdman — CHANGES_REQUESTED** [review]

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [reply specification/columns/tags.md:25]: I'm happy with everything else, but interested in your thinking here

Comment (cnharris10) [reply specification/columns/tags.md]: Sorry, just a mix up from the previous definition.  Fixed now.

**Review by AWS-ZachErdman — APPROVED** [review]
Looks great!

**Comment by jpradocueva** [comment]
Approved by the group on Mar 14.

File: specification/columns/tags.md
Comment (flanakin) [inline specification/columns/tags.md:14]: Can we make sure we have an ack from practitioners who use labels that this is desirable? Just want to confirm a Boolean value is what they want/expect.

**Review by flanakin — APPROVED** [review]

File: specification/columns/tags.md
Comment (udam-f2) [inline specification/columns/tags.md:25]:
```suggestion
This example illustrates three different tagging scenarios. The first two are illustrate when the provider supports both keys and values, while the third is for supporting keys only. The first tag is user-defined and doesn't have a provider prefix. The second tag is provider-defined and has a prefix of `acme/`, which is reserved by the provider. The third tag has a tag key of `baz` and its value is assigned the boolean value `true` since the tag doesn't support a value.
```

The first sentence was a bit hard to read for me. Trying a different way of writing this - accept only if it makes a significant improvement.

File: specification/columns/tags.md
Comment (udam-f2) [inline specification/columns/tags.md:13]:
```suggestion
* A Tag key with a null value for a given resource MAY be included in the tags column depending on the provider's tag finalization process.
```

**Review by udam-f2 — APPROVED** [review]
Minor editorial suggestions for consideration by the group.

File: specification/columns/tags.md
Comment (rupagcp) [reply specification/columns/tags.md]: +1, would like to see what labels and tags looks like on a single line item as well

**Comment by jpradocueva** [comment]
This PR will be put forward for approval on Thursday, March 21.

File: specification/columns/tags.md
Comment (AWS-ZachErdman) [reply specification/columns/tags.md:25]: I'm happy with Udam's suggestion.

File: specification/columns/tags.md
Comment (udam-f2) [reply specification/columns/tags.md]: Example for a label now available

File: specification/columns/tags.md
Comment (flanakin) [inline specification/columns/tags.md:14]: We need to update the supporting content to account for where we're landing with labels. Currently, we have 2 questions documented that feel outdated:
- What does it look like interpreting labels using Excel formulas only? Would users be required to dl a plugin, use vbscript, etc.
- Create a poll around calling the dimension “Tags” or “Labels”?

File: specification/columns/tags.md
Comment (flanakin) [reply specification/columns/tags.md]: We discussed in the call and had a small ack. If anyone is concerned, please leave a comment or create a new issue.

File: specification/columns/tags.md
Comment (udam-f2) [reply specification/columns/tags.md]: I put this up for review at the member meeting today and folks did not vote for changing it. Let's open a separate issue if we want to still make this change - but for now, we're moving forward without to get this long-running PR checked in.

File: specification/columns/tags.md
Comment (udam-f2) [reply specification/columns/tags.md]: Michael said at the maintainer meeting he's good with the logic that this is following. Resolving.

**Pull request merged**
